yt/geometry/particle_geometry_handler.py

```diff
@@ -89,7 +89,12 @@ def _setup_filenames(self):
             else:
                 end = None
             while True:
-                df = cls(self.dataset, self.io, template % {"num": i}, fi, (start, end))
+                try:
+                    df = cls(
+                        self.dataset, self.io, template % {"num": i}, fi, (start, end)
+                    )
+                except FileNotFoundError:
```

Comment:
note that other frontends will directly benefit from this without changes because they call `open` or `h5py.File`, which both raise `FileNotFoundError` naturally.